plat/hisilicon/hikey/hikey_security.c

```diff
@@ -71,10 +71,18 @@ static void sec_protect(uint32_t region_base, uint32_t region_size,
 	volatile struct rgn_attr_reg *rgn_attr;
 	uint32_t i = 0;
 
-	assert(region < 1 || region > 15);
-	assert(!IS_POWER_OF_TWO(region_size) || region_size < 0x10000);
-	/* ensure secure region_base is aligned to region_size */
-	assert((region_base & (region_size - 1)));
+	if (region < 1 || region > 15) {
```

Comment:
@vchong 
I'd recommend that you revert to just using an assert (just with the correct conditions) - the ERROR logging, which would generally be debug is fairly redundant; the assert message would be sufficient to narrow down the issue.

Similar comments for the other checks below.